components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtension.kt

```diff
@@ -81,6 +84,53 @@ abstract class WebExtension(
      * null if port is from a background script.
      */
     abstract fun disconnectPort(name: String, session: EngineSession? = null)
+
+    /**
+     * Registers an [ActionHandler] for this web extension. The handler will
+     * be invoked whenever browser and page action defaults change. To listen
+     * for session-specific overrides see registerActionHandler(
+     * EngineSession, ActionHandler).
+     */
+    abstract fun registerActionHandler(actionHandler: ActionHandler)
```

Comment:
This appears to be registered for the lifecycle of an application. Is there a reason behind that?

Reply:
So, it's registered globally on the webextension, yes. Later I think we'll remove it when extensions are disabled or uninstalled. We're not doing any UI work in these handlers, just making sure the store is up-to-date.

The UI components observing on the store (e.g. your `WebExtensionsToolbarFeature`) are all tied to a lifecycle (e.g. the BrowserFragment). So, we should be good here.